Modules/src/UserInputValidator.cc

```diff
@@ -18,14 +20,29 @@ namespace ChimeraTK {
     if(!change.isValid()) {
       return validateAll();
     }
+
+    if(!_finalised) {
+      throw ChimeraTK::logic_error("Initial values were not validated");
+    }
+
+    // We have downstream channels that signalized a change - invalidate all of our
+    if(_downstreamInvalidatingReturnChannels.count(change) > 0) {
+      _module->setCurrentVersionNumber({});
```

Comment:
(Commenting on a long-merged PR)

Do you remember why (or even whether) this line was necessary? It seems to cause a race condition which is even visible in one of the tests...